cmd/record.go

```diff
@@ -38,15 +38,16 @@ func readRecordConfig() (*models.Record, error) {
 	return &doc.Record, nil
 }
 
-func (t *Record) GetRecordConfig(path *string, proxyPort *uint32, appCmd *string, appContainer, networkName *string, Delay *uint64, passThorughPorts *[]uint) {
-	if isExist := utils.CheckFileExists(filepath.Join(".", "keploy-config.yaml")); !isExist {
+func (t *Record) GetRecordConfig(path *string, proxyPort *uint32, appCmd *string, appContainer, networkName *string, Delay *uint64, passThorughPorts *[]uint, configPath string) error {
+	configFilePath := filepath.Join(configPath, "keploy-config.yaml")
+	if isExist := utils.CheckFileExists(configFilePath); !isExist {
 		t.logger.Info("keploy configuration file not found")
-		return
+		return nil
 	}
-	confRecord, err := readRecordConfig()
+	confRecord, err := readRecordConfig(configFilePath)
 	if err != nil {
 		t.logger.Error("failed to get the record config from config file")
-		return
+		return err
```

Comment:
I gave it a thought initially.

The problem here is `getrecordconfig` can return an error or a nil. I can compare to check if err != nil just after function call and log out the error. But when err is nil, then i would also need to return a string to distinguish whether it was "continuing without keploy configuration file" or "keploy configuration file found", if it is the former then log out the error.
 
refer the return type of the function inside first point of https://go.dev/doc/tutorial/handle-errors to get a clear picture.

do you want to me to this add this instead. It would be a less messy ofc but it could complicate the main function a little bit with if else condition ?

Reply:
You can actually create custom errors like if getRecordConfig gives error something like FileNotFound(custom error) we can identify it like

```
err != nil {
    err == FileNotFound {
       log.info("continuing without keploy")
     }
}
```
We always thought creating customs error in keploy but we never initiated it . You can this out.

Reply:
@charankamarapu does this mean i need to create a new file for errors like this https://github.com/thanos-io/thanos/blob/c74a050a190486addc1ea1ca4b522462fc7ec680/pkg/errors/errors.go ?

Reply:
done!